core/modules/replica/SqlJob.cc

```diff
@@ -192,4 +216,68 @@ void SqlJob::onRequestFinish(SqlRequest::Ptr const& request) {
     }
 }
 
+
+vector<vector<string>> SqlJob::distributeTables(vector<string> const& allTables,
+                                                size_t numBins) {
+
+    // If the total number of tables if less than the number of bins
+    // then we won't be constructing empty bins.
+    vector<vector<string>> tablesPerBin(min(numBins, allTables.size()));
+
+    // The trivial 'round-robin' 
+    for (size_t i=0; i<allTables.size(); ++i) {
+        auto const bin = i % tablesPerBin.size();
+        tablesPerBin[bin].push_back(allTables[i]);
+    }
+    return tablesPerBin;
+}
+
+
+vector<string> SqlJob::workerTables(string const& worker,
+                                    string const& database,
+                                    string const& table) const {
+    vector<string> tables;
+    if (_isPartitioned(database, table)) {
+
+        // The prototype table for creating chunks and chunk overlap tables
+        tables.push_back(table);
+
+        // Locate all chunks registered on the worker. These chunks will be used
+        // to build names of the corresponding chunk-specific partitioned tables.
+        vector<ReplicaInfo> replicas;
+        controller()->serviceProvider()->databaseServices()->findWorkerReplicas(
+            replicas, worker, database);
+
+        for (auto&& replica: replicas) {
+            auto const chunk = replica.chunk();
+            tables.push_back(table + "_" + to_string(chunk));
+            tables.push_back(table + "FullOverlap_" + to_string(chunk));
```

Comment:
Agree. Since the current entry is not the only one which requires attention then it will be a subject for a module-wide refactoring registered in the TODO list: https://confluence.lsstcorp.org/display/DM/Completed%2C+on-going+and+proposed+improvements